### PR TITLE
tests: validation sets spread test

### PR DESCRIPTION
**Pull request by stolowski**
Spread test for validation sets using local assertion. Also fixes a bug with state locking when calling the store.

### CODE REVIEW

File: daemon/api_validate.go [hunk 1]
```diff
@@ -406,7 +406,9 @@ func getSingleSeqFormingAssertion(st *state.State, accountID, name string, seque
 	}
 
 	sequenceKey := []string{release.Series, accountID, name}
+	st.Unlock()
```
Comment (pedronis) [inline daemon/api_validate.go:409]: we should probably add a pokeStateLock to  func (s *apiValidationSetsSuite) SeqFormingAssertion which should have caught this

**Review by pedronis — APPROVED** [review]
thanks,  suggestion for change that should have caught the locking problem

File: daemon/api_validate.go (hunk 1 above)
Comment (stolowski) [reply daemon/api_validate.go:409]: Added, thanks.

File: tests/main/snap-validate-basic/vs1.assert [hunk 2]
```diff
@@ -0,0 +1,29 @@
+type: validation-set
+authority-id: xSfWKGdLoQBoQx88vIM1MpbFNMq53t1f
+series: 16
+account-id: xSfWKGdLoQBoQx88vIM1MpbFNMq53t1f
+name: bar
+sequence: 1
+snaps:
```
Comment (bboozzoo) [inline tests/main/snap-validate-basic/vs1.assert:7]: Could we add a snap with optional presence?

**Review by bboozzoo — APPROVED** [review]
LGTM

File: tests/main/snap-validate-basic/vs1.assert (hunk 2 above)
Comment (stolowski) [reply tests/main/snap-validate-basic/vs1.assert:7]: Yep, I can add it, just found it mildly interesting for the test ;)

Comment (stolowski) [reply tests/main/snap-validate-basic/vs1.assert:7]: Added

**Pull request merged**